src/iminuit/qtwidget.py

```diff
@@ -0,0 +1,413 @@
+                return
+            self.vmax = tmax
+            self.slider.blockSignals(True)
+            if tmax < self.val:
+                self.val = tmax
+                minuit.values[self.par] = tmax
+                self.slider.setValue(int(1e8))
+                self.value_label.setText(f"{self.val:.3g}")
+                self.callback()
+            else:
+                self.slider.setValue(self._float_to_int(self.val))
+            self.slider.blockSignals(False)
+            lim = minuit.limits[self.par]
+            minuit.limits[self.par] = (lim[0], tmax)
+
+        def on_fix_toggled(self):
+            minuit.fixed[self.par] = self.fix.isChecked()
+            if self.fix.isChecked():
+                self.fit.setChecked(False)
+
+        def on_fit_toggled(self):
+            self.slider.setEnabled(not self.fit.isChecked())
+            if self.fit.isChecked():
+                self.fix.setChecked(False)
+            self.callback()
+
+        def reset(self, val, limits=None):
+            if limits is not None:
+                vmin, vmax = limits
+                step = _guess_initial_step(val, vmin, vmax)
+                self.vmin = vmin if np.isfinite(vmin) else val - 100 * step
+                self.vmax = vmax if np.isfinite(vmax) else val + 100 * step
+                self.tmin.blockSignals(True)
+                self.tmin.setValue(self.vmin)
+                self.tmin.blockSignals(False)
+                self.tmax.blockSignals(True)
+                self.tmax.setValue(self.vmax)
+                self.tmax.blockSignals(False)
+
+            self.val = val
+            if self.val < self.vmin:
+                self.vmin = self.val
+                self.tmin.blockSignals(True)
+                self.tmin.setValue(self.vmin)
+                self.tmin.blockSignals(False)
+            elif self.val > self.vmax:
+                self.vmax = self.val
+                self.tmax.blockSignals(True)
+                self.tmax.setValue(self.vmax)
+                self.tmax.blockSignals(False)
+
+            self.slider.blockSignals(True)
+            self.slider.setValue(self._float_to_int(self.val))
+            self.slider.blockSignals(False)
+            self.value_label.setText(f"{self.val:.3g}")
+
+    class MainWindow(QtWidgets.QMainWindow):
+        def __init__(self):
+            super().__init__()
+            self.resize(1200, 600)
+            font = QtGui.QFont()
+            font.setPointSize(12)
+            self.setFont(font)
+            centralwidget = QtWidgets.QWidget(parent=self)
+            self.setCentralWidget(centralwidget)
+            central_layout = QtWidgets.QVBoxLayout(centralwidget)
+            tab = QtWidgets.QTabWidget(parent=centralwidget)
+            interactive_tab = QtWidgets.QWidget()
+            tab.addTab(interactive_tab, "Interactive")
+            results_tab = QtWidgets.QWidget()
+            tab.addTab(results_tab, "Results")
+            central_layout.addWidget(tab)
```

Comment:
You don't need to make a mainwindow here, because you don't use menu bars, a QWidget is enough, I believe. I also don't like that you put the result and the controls onto different tabs. I would like to have them side by side like in the ipywidget.

Reply:
The results are now below the plot. When the fit only has few parameters, it looks kinda stupid though. So if you can think of a better layout, let me know.

Reply:
Please use the same layout like the ipywidget. Use a QHBoxLayout to put the plot left of the controls. You can nest layouts.

Reply:
I don't understand what you mean exactly: the controls are still to the right of the plot:
![Screenshot From 2024-12-11 23-34-39](https://github.com/user-attachments/assets/67c31458-6743-4b06-b140-0bd943db6441)
I am not sure about the layout regarding the results box below, which is not part of the ipywidget, right? So for that I can't use the same layout as the ipywidget. Of course instead of a grid I can use nested vertical and horizontal layouts, but I don't understand what exactly you want changed.

Reply:
Ah ok, now I understand what you mean by fit results. I have never tried your code!

In the ipywidget, I don't show the fit result, because it is largely redundant, the parameter values are shown twice and whether the fit succeeded is shown with the text in the top right. In the interactive fit, showing Hesse errors is not very useful. I suggest to remove the block with the fit result.

Reply:
Ok, I removed that box. Are you okay with the way I organized the parameters? I did not want the window to get super wide, but still have a decent size for the control elements. If there are a lot of parameters, the parameter box will become scrollable.

Reply:
Yeah, that is great, well done. If it is possible I will adapt the ipywidget to look more like this.

Reply:
I also like the colors for the fit and reset buttons

Reply:
Here is another idea as a compromise. The "Minuit.fmin" object is also displayable, it will render as the part above the parameter block. It might be a good idea to show that under the plot and the remove the "success" field on top right in the plot. If we decide that this looks good here, I would also change the ipywidget to follow that design.

Reply:
Ok, there is some empty space there, but I like having fmin displayed there:
![Screenshot From 2024-12-13 20-32-39](https://github.com/user-attachments/assets/507c0332-35a3-4d92-b66d-6abd496f3b1b)

Reply:
Me, too, looks great and is much more intuitive.